aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -223,21 +275,24 @@ Tensor NestedTensor_layer_norm(
     auto acc_type = at::toAccumulateType(input_buffer.scalar_type(), true);
     options = options.dtype(acc_type);
   }
-  Tensor mean = at::empty({valid_word_num}, options);
-  Tensor rstd = at::empty({valid_word_num}, options);
+  Tensor mean = at::empty({M}, options);
+  Tensor rstd = at::empty({M}, options);
   LayerNormKernel(
       input_buffer.is_cuda() ? kCUDA : kCPU,
       input_buffer,
       *weight_contig,
       *bias_contig,
-      valid_word_num,
-      last_dim,
+      M,
+      N,
       eps,
       &output_buffer,
       &mean,
       &rstd);
-  return at::detail::make_tensor<NestedTensorImpl>(
-      std::move(output_buffer), nt_input->get_nested_size_tensor());
+  return std::make_tuple(
+    at::detail::make_tensor<NestedTensorImpl>(std::move(output_buffer), nt_input->get_nested_size_tensor()),
```

Comment:
There are other functions that are preferred to make_tensor. Specifically wrap_buffer:
https://github.com/pytorch/pytorch/blob/089a64e99e2d2b937c72e25c0fa6e4f673b8a1a1/aten/src/ATen/native/nested/NestedTensorUtils.h#L19